src/components/map/map.tsx

```diff
@@ -0,0 +1,234 @@
+import React, { useRef, useEffect, useState, useCallback } from 'react';
+import mapboxgl from 'mapbox-gl';
+import style from './map.module.css';
+import 'mapbox-gl/dist/mapbox-gl.css';
+import { mapboxData } from '@atb/modules/org-data';
+import { isFeaturePoint, isStopPlace } from './utils';
+import { ColorIcon } from '@atb/assets/color-icon';
+import Button from '../button/button';
+import { MonoIcon } from '@atb/assets/mono-icon';
+import { ComponentText, useTranslation } from '@atb/translations';
+
+export type LngLatPosition = [number, number];
+export type MapProps = {
+  position?: LngLatPosition;
+  layer?: string;
+  onSelectStopPlace: (id: string) => void;
+};
+
+type Marker = HTMLElement &
+  React.DetailedReactHTMLElement<
+    {
+      className: string;
+    },
+    HTMLElement
+  >;
```

Comment:
Hmm. Why is this so advanced? Could it be as simple as `JSX.IntrinsicElements['span']`?

Reply:
I meant to ask about this, but forgot. The constructor for the mapboxgl Marker expects an element of type `HTMLElement` and I'll get a ts error using `JSX.IntrinsicElements['span']` as there are a lot more properties to that type 🤔

Reply:
Are you sure this even works? 😅 I have a hard time seeing how this would render properly. CreateElement here returns an JSON structure which react in turns renders as DOM operations at some point. But passing this to setMarker API of mapbox has no idea what to do with that JSON structure. Also the Icon path looks to be wrong?

Reply:
You are absolutely correct, this isn't working 🤦‍♂️ I'll fix it asap.

Reply:
I've solved this using React createRoot. I also added a new component, `PinSvg`, so that I can use the CSS fill property to set the correct color on the SVG.

Reply:
You could also do this the old fashioned way:

```ts
const pin: HTMLElement = document.createElement('img');
pin.src = '/assets/map/Pin.svg';
setMarker(pin);
```

Reply:
That's neat! The design says that we should use the `interactive_0` color for the pin and that's not straightforward to use when using an `<img />` though, is it? 🤔

Reply:
Hm.. No, but I think maybe the color of the pin isn't as important and might not be a hard require. Often in these cases I'd verify and think it over with the designer. Because it's not really interacteble and interactive_0 implies primary function. Also black is more contrast then blue interactive on some blue water for instance. So I think using map/Pin.svg is a better choice here, but like I said, I think verifying/discussing this is a good way to go

Reply:
That's true. Good point. I'll have a chat with the designer 👍

Reply:
I've talked with the designer and we'll go for the black SVG. Updated the code to use `document.createElement` 👍